external/js/cothority/src/network/connection.ts

```diff
@@ -113,15 +113,15 @@ export class WebSocketConnection implements IConnection {
             });
 
             ws.onClose((code: number, reason: string) => {
-                if (code !== 1000) {
-                    Log.error("Got close:", code, reason);
-                    reject(new Error(reason));
+                if (code === 1000) {
+                    Log.lvl3("Normal closing of connection");
```

Comment:
There's a `return` missing here

Reply:
oups - nice catch. Wonder why it passed the tests?

Reply:
onClose so I guess the resolve is called before

Reply:
so if the `resolve` is called, a later `reject` doesn't work anymore? Makes sense.